lib/new_relic/agent/instrumentation/rainbows_instrumentation.rb

```diff
@@ -12,6 +12,17 @@
   executes do
     ::NewRelic::Agent.logger.info 'Installing Rainbows instrumentation'
     ::NewRelic::Agent.logger.info 'Detected Rainbows, please see additional documentation: https://newrelic.com/docs/troubleshooting/im-using-unicorn-and-i-dont-see-any-data'
+
+    deprecation_msg = 'The dispatcher rainbows is deprecated. It will be removed ' \
+     'in version 9.0.0. Please use a supported dispatcher instead. ' \
+     'Visit https://docs.newrelic.com/docs/apm/agents/ruby-agent/getting-started/ruby-agent-requirements-supported-frameworks for options.'
+
+    ::NewRelic::Agent.logger.log_once(
+      :warn,
+      :deprecated_rainbows_dispatcher,
+      deprecation_msg
+    )
+    ::NewRelic::Agent.record_metric("Supportability/Deprecated/Rainbows", 1)
```

Comment:
> Like with climate change, you know? It used to rain here all of the time and now it's almost all gone. I don't even know if it's raining somewhere else, or if it's dry there too. Dry prolly. And that's when it hit me. These rainbows are the last ones, man. They won't be here for much longer. So that's when I decided to name the band 'Deprecated Rainbows'.